staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_test.go

```diff
@@ -2269,3 +2261,136 @@ func TestLoadObjects(t *testing.T) {
 		})
 	}
 }
+
+func TestApplyObjects(t *testing.T) {
+	testdirs := []string{"testdata/prune/simple"}
+	for _, testdir := range testdirs {
+		t.Run(testdir, func(t *testing.T) {
+			k := fakekube.NewFakeKube(t)
+			f := cmdutil.NewFactory(k.AsRESTClientGetter())
+
+			cmdtesting.WithAlphaEnvs([]cmdutil.FeatureGate{cmdutil.ApplySet}, t, func(t *testing.T) {
+				manifests := []string{"manifest1", "manifest2"}
+				for _, manifest := range manifests {
+					t.Logf("applying manifest %v", manifest)
+
+					cmd := &cobra.Command{}
+					flags := NewApplyFlags(genericclioptions.NewTestIOStreamsDiscard())
+					flags.AddFlags(cmd)
+					cmd.Flags().Set("filename", filepath.Join(testdir, manifest+".yaml"))
+					cmd.Flags().Set("applyset", filepath.Base(testdir))
+					cmd.Flags().Set("prune", "true")
+					cmd.Flags().Set("validate", "false")
+
+					o, err := flags.ToOptions(f, cmd, "kubectl", []string{})
+					if err != nil {
+						t.Fatalf("unexpected error creating apply options: %v", err)
+					}
+
+					err = o.Validate()
+					if err != nil {
+						t.Fatalf("unexpected error from validate: %v", err)
+					}
+
+					var unifiedOutput bytes.Buffer
+					o.Out = &unifiedOutput
+					o.ErrOut = &unifiedOutput
+
+					if err := o.Run(); err != nil {
+						t.Errorf("error running apply: %v", err)
+					}
+
+					got := unifiedOutput.String()
+
+					p := filepath.Join(testdir, manifest+"-expected-apply.txt")
+					wantBytes, err := os.ReadFile(p)
+					if err != nil {
+						t.Fatalf("error reading file %q: %v", p, err)
+					}
+					want := string(wantBytes)
+					if diff := cmp.Diff(want, got); diff != "" {
+						t.Errorf("apply output has unexpected diff (-want +got):\n%s", diff)
+					}
+				}
+			})
+		})
+	}
+}
+
+func TestApplyFail(t *testing.T) {
+	testdirs := []string{"testdata/prune/simple"}
+	for _, testdir := range testdirs {
+		t.Run(testdir, func(t *testing.T) {
+			k := fakekube.NewFakeKube(t)
```

Comment:
My only concern is this new fake kube helper, and as much as I hate the TestFactory, I'm inclined to ask what's missing there that you couldn't re-use it? Looking at https://github.com/kubernetes/kubernetes/blob/44bedc2a46f7403bebb1c9ce019d7f6da0084908/staging/src/k8s.io/kubectl/pkg/cmd/delete/delete_test.go#L50-L99 I'm inclined to say that what you're doing below is achivable in roughly similar size of code.

Reply:
I'll give it a try in an alternative PR, I think when I tried before I was having to mock more clients than just this, but I forget the details so will put together another PR...  it does look like I still have to parse the REST requests, it's just not reusable across tests, but I will give it a go.

Reply:
Alternative PR is #116205